replica/table.cc

```diff
@@ -475,7 +470,9 @@ const lw_shared_ptr<sstables::sstable_set>& compaction_group::main_sstables() co
 }
 
 void compaction_group::set_main_sstables(lw_shared_ptr<sstables::sstable_set> new_main_sstables, std::optional<int64_t> delta_disk_space_used) {
-    _main_sstables = std::move(new_main_sstables);
+    // Readers hold a shared_ptr of the sstable_set_impl
+    // so it's safe to update the sstable_set
+    *_main_sstables = std::move(*new_main_sstables);
```

Comment:
Doesn't it violate snapshot isolation for the sstable set?

Before the PR, when someone obtained `lw_shared_ptr<sstables::sstable_set>`, it could assume that it doesn't change.

For example:
```
partition_presence_checker
table::make_partition_presence_checker(lw_shared_ptr<sstables::sstable_set> sstables) {
```

It holds to `lw_shared_ptr<sstables::sstable_set>`, not the impl.